sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -111,11 +125,11 @@ public static BinaryData fromStream(InputStream inputStream) {
     }
```

Comment:
Are we not able to handle this in a more efficient manner? As it stands this means that the entire input stream is read into a buffer, converted into a byte array, and then copied into a new byte array? At the very least you can avoid the byte array copy.

Reply:
Removed the last byte array and same optimization in other from..() API as well.